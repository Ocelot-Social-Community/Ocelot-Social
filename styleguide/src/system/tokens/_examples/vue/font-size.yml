#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
