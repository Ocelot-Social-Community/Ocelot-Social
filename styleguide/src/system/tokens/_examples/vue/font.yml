#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
