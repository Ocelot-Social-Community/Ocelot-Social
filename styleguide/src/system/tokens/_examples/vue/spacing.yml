#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
