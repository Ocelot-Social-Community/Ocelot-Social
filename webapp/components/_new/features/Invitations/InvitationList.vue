<template>
  <div class="invitation-list">
    <ul v-if="validInviteCodes.length">
      <client-only>
        <invitation
          v-for="inviteCode in validInviteCodes"
          :key="inviteCode.code"
          :invite-code="inviteCode"
          :copy-message="copyMessage"
          @invalidate-invite-code="invalidateInviteCode"
        />
      </client-only>
    </ul>
    <div v-else class="no-invitation">
      {{ $t('invite-codes.no-links', { max: maxLinks }) }}
    </div>
    <create-invitation
      @generate-invite-code="generateInviteCode"
      :disabled="isLimitReached"
      class="create-invitation"
    />
  </div>
</template>

<script>
import Invitation from './Invitation.vue'
import CreateInvitation from './CreateInvitation.vue'

export default {
  name: 'InvitationList',
  props: {
    inviteCodes: {
      type: Array,
      required: true,
    },
    copyMessage: {
      type: String,
      default: '',
    },
  },
  components: {
    Invitation,
    CreateInvitation,
  },
  computed: {
    validInviteCodes() {
      return this.inviteCodes.filter((inviteCode) => inviteCode.isValid)
    },
    maxLinks() {
      return Number(this.$env.INVITE_LINK_LIMIT)
    },
    isLimitReached() {
      return this.validInviteCodes.length >= this.maxLinks
    },
  },
  methods: {
    generateInviteCode(comment) {
      this.$emit('generate-invite-code', comment)
    },
    invalidateInviteCode(code) {
      this.$emit('invalidate-invite-code', code)
    },
  },
}
</script>

<style scoped lang="scss">
.invitation-list {
  display: flex;
  flex-flow: column;
  gap: $space-base;
  padding-bottom: $space-base;

  ul {
    list-style: none;
  }
}

.create-invitation {
  margin-top: $space-base;
}
</style>
