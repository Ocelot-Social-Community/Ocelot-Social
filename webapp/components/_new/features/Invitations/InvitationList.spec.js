import { render, screen, fireEvent } from '@testing-library/vue'
import '@testing-library/jest-dom'

import InvitationList from './InvitationList.vue'

const localVue = global.localVue

Object.assign(navigator, {
  clipboard: {
    writeText: jest.fn(),
  },
})

const sampleInviteCodes = [
  {
    code: 'test-invite-code-1',
    comment: 'test-comment',
    redeemedByCount: 0,
    isValid: true,
  },
  {
    code: 'test-invite-code-2',
    comment: 'test-comment-2',
    redeemedByCount: 1,
    isValid: true,
  },
  {
    code: 'test-invite-code-3',
    comment: 'test-comment-3',
    redeemedByCount: 0,
    isValid: false,
  },
]

describe('InvitationList.vue', () => {
  let wrapper

  const Wrapper = ({ withInviteCodes, withCopymessage = false, limit = 3 }) => {
    const propsData = {
      inviteCodes: withInviteCodes ? sampleInviteCodes : [],
      copyMessage: withCopymessage ? 'test-copy-message' : undefined,
    }
    return render(InvitationList, {
      localVue,
      propsData,
      mocks: {
        $t: jest.fn((v) => v),
        $toast: {
          success: jest.fn(),
          error: jest.fn(),
        },
        $env: {
          INVITE_LINK_LIMIT: limit,
        },
      },
      stubs: {
        'client-only': true,
      },
    })
  }

  it('renders', () => {
    wrapper = Wrapper({ withInviteCodes: true })
    expect(wrapper.container).toMatchSnapshot()
  })

  it('renders empty state', () => {
    wrapper = Wrapper({ withInviteCodes: false })
    expect(wrapper.container).toMatchSnapshot()
  })

  it('does not render invalid invite codes', () => {
    wrapper = Wrapper({ withInviteCodes: true })
    const invalidInviteCode = screen.queryByText('invite-codes.test-invite-code-3')
    expect(invalidInviteCode).not.toBeInTheDocument()
  })

  describe('without copy message', () => {
    beforeEach(() => {
      wrapper = Wrapper({ withCopymessage: false, withInviteCodes: true })
    })

    it('can copy a link', async () => {
      const clipboardMock = jest.spyOn(navigator.clipboard, 'writeText').mockResolvedValue()
      const copyButton = screen.getAllByLabelText('invite-codes.copy-code')[0]
      await fireEvent.click(copyButton)
      expect(clipboardMock).toHaveBeenCalledWith(
        'http://localhost/registration?method=invite-code&inviteCode=test-invite-code-1',
      )
    })
  })

  describe('with copy message', () => {
    beforeEach(() => {
      wrapper = Wrapper({ withCopymessage: true, withInviteCodes: true })
    })

    it('can copy the link with message', async () => {
      const clipboardMock = jest.spyOn(navigator.clipboard, 'writeText').mockResolvedValue()
      const copyButton = screen.getAllByLabelText('invite-codes.copy-code')[0]
      await fireEvent.click(copyButton)
      expect(clipboardMock).toHaveBeenCalledWith(
        'test-copy-message http://localhost/registration?method=invite-code&inviteCode=test-invite-code-1',
      )
    })
  })

  it('cannot generate more than the limit of invite codes', () => {
    wrapper = Wrapper({ withInviteCodes: true, limit: 3 })
    const generateButton = screen.getByText('invite-codes.generate-invite-code')
    expect(generateButton).toBeDisabled()
  })
})
