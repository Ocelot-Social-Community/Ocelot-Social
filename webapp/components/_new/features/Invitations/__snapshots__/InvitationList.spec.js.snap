// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitationList.vue renders 1`] = `
<div>
  <div
    class="invitation-list"
  >
    <ul>
      <client-only-stub>
        <li
          class="invitation"
        >
          <div
            class="column1"
          >
            <div
              class="code"
            >
              
      test-invite-code-1
      
              <span
                class="mdash"
              >
                —
              </span>
               
              <span
                class="comment"
              >
                test-comment
              </span>
            </div>
             
            <div>
              <span>
                
        invite-codes.redeemed-count-0
      
              </span>
            </div>
          </div>
           
          <div
            class="actions"
          >
            <button
              aria-label="invite-codes.copy-code"
              class="copy-button base-button --icon-only --circle"
              type="button"
            >
              <span
                class="base-icon"
              >
                <!---->
              </span>
               
              <!---->
               
            </button>
             
            <button
              aria-label="invite-codes.invalidate"
              class="invalidate-button base-button --icon-only --circle"
              type="button"
            >
              <span
                class="base-icon"
              >
                <!---->
              </span>
               
              <!---->
               
            </button>
          </div>
        </li>
        <li
          class="invitation"
        >
          <div
            class="column1"
          >
            <div
              class="code"
            >
              
      test-invite-code-2
      
              <span
                class="mdash"
              >
                —
              </span>
               
              <span
                class="comment"
              >
                test-comment-2
              </span>
            </div>
             
            <div>
              <span>
                
        invite-codes.redeemed-count
      
              </span>
            </div>
          </div>
           
          <div
            class="actions"
          >
            <button
              aria-label="invite-codes.copy-code"
              class="copy-button base-button --icon-only --circle"
              type="button"
            >
              <span
                class="base-icon"
              >
                <!---->
              </span>
               
              <!---->
               
            </button>
             
            <button
              aria-label="invite-codes.invalidate"
              class="invalidate-button base-button --icon-only --circle"
              type="button"
            >
              <span
                class="base-icon"
              >
                <!---->
              </span>
               
              <!---->
               
            </button>
          </div>
        </li>
      </client-only-stub>
    </ul>
     
    <div
      class="create-invitation create-invitation"
    >
      <div>
        invite-codes.generate-code-explanation
      </div>
       
      <form
        class="generate-invite-code-form"
      >
        <div
          class="ds-form-item ds-input-size-base"
        >
          <label
            class="ds-input-label"
            style="display: none;"
          >
            
  

          </label>
          <div
            class="ds-input-wrap"
          >
            <!---->
            <input
              class="ds-input"
              name="comment"
              placeholder="invite-codes.comment-placeholder"
              tabindex="0"
              type="text"
            />
            <!---->
          </div>
          <transition-stub
            name="ds-input-error"
          >
            <div
              class="ds-input-error"
              style="display: none;"
            >
              
    
  
            </div>
          </transition-stub>
        </div>
         
        <button
          aria-label="invite-codes.generate-code"
          class="generate-invite-code base-button --icon-only --circle"
          type="submit"
        >
          <span
            class="base-icon"
          >
            <!---->
          </span>
           
          <!---->
           
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`InvitationList.vue renders empty state 1`] = `
<div>
  <div
    class="invitation-list"
  >
    <div
      class="no-invitation"
    >
      
    invite-codes.no-links
  
    </div>
     
    <div
      class="create-invitation create-invitation"
    >
      <div>
        invite-codes.generate-code-explanation
      </div>
       
      <form
        class="generate-invite-code-form"
      >
        <div
          class="ds-form-item ds-input-size-base"
        >
          <label
            class="ds-input-label"
            style="display: none;"
          >
            
  

          </label>
          <div
            class="ds-input-wrap"
          >
            <!---->
            <input
              class="ds-input"
              name="comment"
              placeholder="invite-codes.comment-placeholder"
              tabindex="0"
              type="text"
            />
            <!---->
          </div>
          <transition-stub
            name="ds-input-error"
          >
            <div
              class="ds-input-error"
              style="display: none;"
            >
              
    
  
            </div>
          </transition-stub>
        </div>
         
        <button
          aria-label="invite-codes.generate-code"
          class="generate-invite-code base-button --icon-only --circle"
          type="submit"
        >
          <span
            class="base-icon"
          >
            <!---->
          </span>
           
          <!---->
           
        </button>
      </form>
    </div>
  </div>
</div>
`;
